src/pipelines/stage.ts

```diff
@@ -42,16 +51,90 @@ export abstract class DataStage extends Stage {
   }
 
   addAlarm(id: string, props: AlarmProps): DataStage {
-    this.cloudwatchAlarms.push(new cloudwatch.Alarm(this, id, {
-      metric: props.metric,
-      comparisonOperator: props.comparisonOperator ?? cloudwatch.ComparisonOperator.GREATER_THAN_THRESHOLD,
-      threshold: props.threshold ?? 5,
-      evaluationPeriods: props.evaluationPeriods ?? 1,
-    }));
+    this.cloudwatchAlarms.push(
+      new cloudwatch.Alarm(this, id, {
+        metric: props.metric,
+        comparisonOperator:
+          props.comparisonOperator ??
+          cloudwatch.ComparisonOperator.GREATER_THAN_THRESHOLD,
+        threshold: props.threshold ?? 5,
+        evaluationPeriods: props.evaluationPeriods ?? 1,
+      }),
+    );
     return this;
   }
 }
 
+export abstract class StateMachineStage extends DataStage {
+  readonly targets?: events.IRuleTarget[];
+  readonly eventPattern?: events.EventPattern;
+
+  public stateMachine?: sfn.StateMachine;
+  public stateMachineInput?: { [key: string]: any };
```

Comment:
These two don't need to be optional, do they? A state machine will always be created. As far the input is concerned, if we don't get it in the props, we would just set it to `{}`.

Reply:
`stateMachine` should be required, changing that, but `stateMachineInput` should allow for undefined/null value.

Reply:
When a null value is given for `stateMachineInput`, doesn't this mean we should pass an empty object to the state machine? I'm not sure what exactly happens to `RuleInput.fromObject(stateMachineInput)` when `stateMachineInput` is null.

Reply:
I follow you now `{}` vs `undefined`